mathesar/api/serializers/records.py

```diff
@@ -2,11 +2,25 @@
 
 
 class RecordListParameterSerializer(serializers.Serializer):
-    filters = serializers.JSONField(required=False, default=[])
+    filters = serializers.JSONField(required=False, default=None)
     order_by = serializers.JSONField(required=False, default=[])
-    group_count_by = serializers.JSONField(required=False, default=[])
+    duplicate_only = serializers.JSONField(required=False, default=None)
+    grouping = serializers.JSONField(required=False, default={})
+
+    def validate_duplicate_only(self, duplicate_only):
+        is_none = duplicate_only is None
+        if not is_none:
+            is_a_list = type(duplicate_only) is list
+            if not is_a_list:
```

Comment:
```suggestion
            if type(duplicate_only) is not list:
```

Reply:
This change taxes the reader since he has to parse the boolean expression to tell what it checks. Are you optimizing for terseness?

Reply:
By the way, I tend to not declare negative booleans, since negation is a discreet cognitive step and treating it as a discreet modifier is more congruent with that, hence `not is_a_list` instead of `is_not_a_list`.

Reply:
I actually find that it's easier to read/parse this way. I don't have to keep track of a variable that's not used anywhere else and it's clear what the conditional is actually checking. The variable name `is_a_list` doesn't let me know _what_ is a list or not. 

One of my favorite things about Python specifically is that the syntax easy to read in terms of English sentences. You can see that with `if type(duplicate_only) is not list:` --> "if the type of duplicate_only is not a list, do blah". You lose that readability with having a separate variable.

Reply:
I can tell that our styles are clashing ⚔️ I'll try to make a strong case for my style, since I find that it suits me very well.

> The variable name is_a_list doesn't let me know what is a list or not.

A central idea behind doing things this way is that it should not. What a list is should be defined separately. For the sake of readability and reuse. It's about being declarative. We're offloading the burden of knowing what a list is (it's something that has the `type` equal to `list`) to a dedicated variable, and a dedicated cognitive step. This would be easier to illustrate with something more complicated, but if we were to change the definition of a list, we'd change the piece of code labeled `is_a_list`, while the `if` block would stay the same. My objective is to split every cognitive step up into discreet code objects. If it were something more complicated, like `type(x) is Sequence and x.__name__ == "dict" and len(x.keys) > 0` (pseudo code), this question would be less nuanced; but, even with a subtle case like `type(duplicate_only) is not list` the same principle applies: we shouldn't care what a list is unless it's relevant. If it's always relevant, then there's conflicting definitions for what a list is and that can be looked at.

This "split every cognitive step up into discreet code objects" principle improves my code reading/reviewing bandwidth significantly.

> `if type(duplicate_only) is not list`: --> "if the type of duplicate_only is not a list, do blah".

```
type(x) is not list:
```
reads as "type of x is not a list", but why do we care that that's `type(x)`'s value? Because, it means that it's a list. That's a subtle cognitive step that's left implicit.

```
not is_a_list
```
reads as "is not a list", and it leaves nothing implicit. Again, this would be more illustrative if the example was more complicated.

`not is_a_list` is a less specific statement. In this case, the type of `x` is an implementation detail that we don't care about. If the way to check whether it's a list were different nothing would change in the context of this method, only the declaration for what a list is, which `type(x) is not list` leaves implicit, and `not is_a_list` (together with declaration of `is_a_list`) makes explicit.

The principle smoothly extends into `not is_something(x)`, for when you want a higher-scope definition for what `something` is.

By the way, I'm surprised that you don't find that `not is_a_list` rolls off the tongue. I too find that being able to easily read code aloud is desirable.